File: src/main/java/com/kairlec/dao/UserMapper.java
package com.kairlec.dao;

import com.kairlec.pojo.User;
import org.apache.ibatis.annotations.*;

import java.util.List;

@Mapper
public interface UserMapper {

    Integer init();

    List<User> getAll();

    User getUser(@Param("username") String username);

    Integer insertUser(User user);

    Integer deleteUser(@Param("username") String username);

    Integer updateLoginInfo(User user);

    Integer updatePassword(User user);
}

File: src/main/java/com/kairlec/dao/UserMapper.kt
package com.kairlec.dao;

import com.kairlec.pojo.User
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface UserMapper {
    fun init(): Int?
    fun getAll(): List<User>?
    fun getUser(@Param("username") username: String): User?
    fun insertUser(user: User): Int?
    fun deleteUser(@Param("username") username: String): Int?
    fun updateLoginInfo(user: User): Int?
    fun updatePassword(user: User): Int?
}
